Review tests for task 0

oscamaina opened the pull request.

oscamaina committed:
Initial project functions layout

oscamaina committed:
initial test cases

oscamaina committed:
Adds more tests

mnjaggah reviewed (commented):
with the assertEqual() method, is the second parameter the second value being compared or the msg 
considering assertEqual(a, b, msg=None)

munala commented on test.py:
```diff
@@ -0,0 +1,69 @@
+    """Test cases for Adding Persons"""
+    def setUp(self):
+        self.dojo = Dojo()
+        self.dojo.fellows = []
```
The fellows list is not in the dojo module. You can add it there.

oscamaina replied:
@lederp23 thanks, i also have to add the staffs list

munala commented on test.py:
```diff
@@ -0,0 +1,69 @@
+    """Test cases for Adding Persons"""
+    def setUp(self):
+        self.dojo = Dojo()
+        self.dojo.fellows = []
```
The fellows list is not in the dojo module. You can add it there.

munala replied:
The fellows list is not in the dojo module. You can add it there.

ianoti replied:
this list should not be initialised in the test setUp, you've created an instance of the class and this will include the list you declared in the class Dojo.

munala commented on test.py:
```diff
@@ -0,0 +1,69 @@
+    def setUp(self):
+        self.dojo = Dojo()
+        self.dojo.fellows = []
+        self.dojo.staffs = []
```
The staff list is not in the dojo module. You can add it there.

oscamaina committed:
initializes lists

oscamaina committed:
adds requirements list

ianoti commented on requirements.txt:
```diff
@@ -0,0 +1,50 @@
+appdirs==1.4.0
```
This file is too large with too many package dependencies. ensure you run the `pip freeze` while the virtualenv is active to remove unnecesary files.

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+    def test_adds_office_successfully(self):
+        """Tests if room of type office can be added successfully"""
+        office = self.dojo.create_room("office", "A04")
+        self.assertEqual(office, "Office added successfully")
```
add an assertion for office count number increasing by 1. If you have a list of offices you can use this to check the number of offices.

ianoti replied:
also add a check that the room created is an instance of `Office`.

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+    def test_adds_living_space_successfully(self):
+        """Tests if room of type livingSpace can be added successfully"""
+        living_space =self.dojo.create_room("living_space", "Arusha")
+        self.assertEqual(living_space, "Living space added successfully")
```
refer to earlier comment on checking the count and that the created room is a `Living space`.

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+    def test_create_wrong_room_type(self):
+        """Tests adding wrong room type """
+        room = self.dojo.create_room("social_hall", "Stam")
+        self.assertEqual(room, "Wrong room type")
```
assert that the number of rooms has not changed to show it hasn't created a room

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+        current_no_fellows = len(self.dojo.fellows)
+        self.dojo.add_person('Kipyegon Ken', 'Fellow')
+        new_no_fellows = len(self.dojo.fellows)
+        self.assertEqual(new_no_fellows, current_no_fellows+1)
```
a better way to write this would be `self.assertEqual(new_no_fellows-current_no_fellows, 1)`.
Also in this instance it's okay to check if the person created is an instance of Fellow class.

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+        current_no_staff = len(self.dojo.staffs)
+        self.dojo.add_person('Shem Ogube', 'Staff')
+        new_no_staff = len(self.dojo.staffs)
+        self.assertEqual(new_no_staff, current_no_staff+1)
```
refer to earlier comment on syntax of the assert statement for checking number of staff

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+        new = self.dojo.add_person("Wekesa Maina", "Bootcamper", "N")
+        self.assertEqual(new,"Wrong input. Can only be FELLOW or STAFF")
+
+    def test_staff_can_not_be_given_accomodation(self):
```
This particular test breaks the system if implemented. Consider removing this test because the system should allow people to be added if there are no rooms, and will assign them rooms later when rooms are available. I would recommend putting this test on hold until you reach implementation of later functionalities like `reallocate`

jomomg commented:
Everything looks OK. I see no issues with the tests.

ianoti commented on test.py:
```diff
@@ -0,0 +1,69 @@
+        staff = self.dojo.add_person("Ochieng Collins", "Staff", "Y")
+        self.assertEqual(staff, "staff cannot be created")
+        
+    def test_add_person_that_already_exist(self):
```
good thinking on this test, but consider that there will be people with the same first and last names. A better way would be to let the system add both people but have an `id` attribute that can be used to differentiate between the two people and assert that these two `id` are different.

ianoti reviewed (commented):
Good job so far, I have suggested some changes you can make to the tests. Consider installing `nose` to run your tests to simplify the test running process. this eliminates the need to have lines `68-69` in your test file code.

oscamaina committed:
removed unnecessary files

oscamaina committed:
updated the requirement list

oscamaina committed:
refactor tests

oscamaina committed:
initial functionality implementation

ianoti commented on dojo.py:
```diff
@@ -0,0 +1,32 @@
+		if room_name in self.rooms:
+			return "Room already exists"
+		else:
+			room_type = input("Enter room type for " + room_name + ":")
```
In your implementation of the logic make use of classes to create room objects with the attributes. Consider changing your implementation logic to make use of classes for the rooms.

ianoti commented on requirements.txt:
```diff
@@ -0,0 +1,5 @@
+appdirs==1.4.2
```
good job updating requirements.txt file

ianoti commented on test.py:
```diff
@@ -0,0 +1,83 @@
+        office = self.dojo.create_room("office", "A04")
+        new_no_offices = len(self.dojo.offices)
+        self.assertEqual(office, "Office added successfully")
+        self.assertEqual(new_no_offices, current_no_offices+1)
```
change the syntax of this assertion to reflect the change I indicated as you did in lines 72 and 65

ianoti commented on test.py:
```diff
@@ -0,0 +1,83 @@
+        living_space =self.dojo.create_room("living_space", "Arusha")
+        new_no_livingSpaces = len(self.dojo.livingSpaces)
+        self.assertEqual(living_space, "Living space added successfully")
+        self.assertEqual(new_no_livingSpaces, current_no_livingSpaces+1)
```
refer to earlier comment on syntax of assertions

ianoti commented on test.py:
```diff
@@ -0,0 +1,83 @@
+        new = self.dojo.add_person("Wekesa Maina", "Bootcamper", "N")
+        self.assertEqual(new,"Wrong category. Can only be fellow or staff")
+        
+    def test_add_person_that_already_exist(self):
```
as I indicated earlier, this test is not ideal for version 0 of the system, it will lock you in to never allowing people with the same name to work where your system is being used. Consider putting this test on hold until you implement unique identification system for persons in the system

ianoti reviewed (commented):
Good job on the corrections for the tests, when implementing functionality remember to make use of OOP concepts. Also keep in mind that you shall implement the code using `docopt` so don't have unnecessary prompts for input in the function.

oscamaina committed:
refactor tests

oscamaina committed:
remodelled method create_room

remodelled create_room method to use rooms model

oscamaina committed:
initial commit for rooms model

oscamaina committed:
initial commit for person model

oscamaina committed:
adds app interface

oscamaina committed:
implemented adding and allocating persons

oscamaina committed:
refactor tests

oscamaina committed:
modified classes

added occupants variable to room class,added capacity to classes office and livingSpace

oscamaina committed:
modified create_room method

modiefied create_room method to enable creating as many
rooms as possible by specifying multiple room names
after the create_room command

oscamaina committed:
refactor tests

fixed bug in method create_room to take a list of room names instead of
strings

oscamaina committed:
refactor tests

refactored to test the for inheritance

oscamaina committed:
modified functionalities

modified the functions to make use of the defined models

oscamaina committed:
moddified usage

updated the commands, modified command add_person to take first_name and
last_name arguments

oscamaina committed:
adds rooms model

oscamaina committed:
adds person model

oscamaina committed:
remove person.py

oscamaina committed:
remove test.py
